csrc/quantization/cutlass_w8a8/grouped_mm_c3x.cuh

```diff
@@ -0,0 +1,188 @@
+    const int64_t a_base_as_int, const int64_t b_base_as_int,
+    const int64_t out_base_as_int, const int64_t a_scales_base_as_int,
+    const int64_t b_scales_base_as_int, int64_t n, int64_t k,
+    bool per_act_token, bool per_out_ch, int64_t ab_size, int64_t c_size,
+    int64_t acc_size) {
+  int expert_id = threadIdx.x;
+
+  int64_t expert_offset = expert_offsets[expert_id];
+
+  a_offsets[expert_id] = a_base_as_int + expert_offset * k * ab_size;
+  b_offsets[expert_id] = b_base_as_int + expert_id * k * n * ab_size;
+  out_offsets[expert_id] = out_base_as_int + expert_offset * n * c_size;
+  a_scales_offsets[expert_id] =
+      a_scales_base_as_int + (per_act_token ? expert_offset : 0) * acc_size;
+  b_scales_offsets[expert_id] =
+      b_scales_base_as_int +
+      (per_out_ch ? n * expert_id : expert_id) * acc_size;
+}
+
+namespace {
+
+using ProblemShape =
+    cutlass::gemm::GroupProblemShape<cute::Shape<int, int, int>>;
+using ElementAB_Type = cutlass::float_e4m3_t;
+using ElementC_Type = cutlass::half_t;
+
+using ElementAccumulator = float;
+using ArchTag = cutlass::arch::Sm90;
+using OperatorClass = cutlass::arch::OpClassTensorOp;
+
+using LayoutA = cutlass::layout::RowMajor;
+using LayoutB = cutlass::layout::ColumnMajor;
+using LayoutC = cutlass::layout::RowMajor;
+
+template <typename ElementAB_, typename ElementC_,
+          template <typename, typename, typename> typename Epilogue_,
+          typename TileShape, typename ClusterShape, typename KernelSchedule,
+          typename EpilogueSchedule>
+struct cutlass_3x_group_gemm {
+  using ElementAB = ElementAB_;
+  using ElementC = void;
+  using ElementD = ElementC_;
+  using ElementAccumulator = float;
+
+  using Epilogue = Epilogue_<ElementAccumulator, ElementD, TileShape>;
+
+  using StrideC =
+      cute::remove_pointer_t<cute::Stride<int64_t, cute::Int<1>, cute::Int<0>>>;
+
+  const int AlignmentAB = 128 / cutlass::sizeof_bits<ElementAB>::value;
+  const int AlignmentC = 128 / cutlass::sizeof_bits<ElementD>::value;
+
+  using EVTCompute = typename Epilogue::EVTCompute;
+
+  using CollectiveEpilogue =
+      typename cutlass::epilogue::collective::CollectiveBuilder<
+          ArchTag, OperatorClass, TileShape, ClusterShape,
+          cutlass::epilogue::collective::EpilogueTileAuto, ElementAccumulator,
+          ElementAccumulator, ElementC, LayoutC*, 4, ElementD, LayoutC*, 4,
+          EpilogueSchedule, EVTCompute>::CollectiveOp;
+
+  static constexpr size_t CEStorageSize =
+      sizeof(typename CollectiveEpilogue::SharedStorage);
+  using Stages = typename cutlass::gemm::collective::StageCountAutoCarveout<
+      static_cast<int>(CEStorageSize)>;
+
+  using CollectiveMainloop =
+      typename cutlass::gemm::collective::CollectiveBuilder<
+          ArchTag, OperatorClass, ElementAB, LayoutA*, 16, ElementAB, LayoutB*,
+          16, ElementAccumulator, TileShape, ClusterShape, Stages,
+          KernelSchedule>::CollectiveOp;
```

Comment:
Similar: Should these 4s be `AlignmentAB`? It looks like `AlignmentAB` isn't used anywhere